core/src/main/scala/org/apache/spark/util/ByteBufferOutputStream.scala

```diff
@@ -23,7 +23,9 @@ import java.nio.ByteBuffer
 /**
  * Provide a zero-copy way to convert data in ByteArrayOutputStream to ByteBuffer
  */
-private[spark] class ByteBufferOutputStream extends ByteArrayOutputStream {
+private[spark] class ByteBufferOutputStream(val size: int) extends ByteArrayOutputStream(size) {
```

Comment:
Although `ByteArrayOutputStream` calls it `size`, I prefer to use `capacity`. And please remove `val` here. E.g.,
```
private[spark] class ByteBufferOutputStream(capacity: Int) extends ByteArrayOutputStream(capacity) {
```